plugins/services/src/js/components/ServiceStatusWarning.js

```diff
@@ -7,29 +7,48 @@ import Icon from '../../../../../src/js/components/Icon';
 import Pod from '../structs/Pod';
 import Service from '../structs/Service';
 import ServiceTree from '../structs/ServiceTree';
+import StringUtil from '../../../../../src/js/utils/StringUtil';
 
 class ServiceStatusWarning extends Component {
+  getTooltip(content) {
+    return (
+      <Tooltip
+        content={content}
+        width={250}
+        wrapText={true}
+        wrapperClassName="tooltip-wrapper status-waiting-indicator">
+        <Icon color="red" id="yield" size="mini" />
+      </Tooltip>
+    );
+  }
+
   getTooltipContent(timeWaiting) {
     return `DC/OS has been waiting for resources and is unable to complete this deployment for ${DateUtil.getDuration(timeWaiting, null)}.`;
   }
+
   render() {
     const {item} = this.props;
-    const queue = item.getQueue();
-
-    if (queue != null
-      && DeclinedOffersUtil.shouldDisplayDeclinedOffersWarning(queue)) {
-      const waitingSince = DateUtil.strToMs(queue.since);
-      const timeWaiting = Date.now() - waitingSince;
-
-      return (
-        <Tooltip
-          content={this.getTooltipContent(timeWaiting)}
-          width={250}
-          wrapText={true}
-          wrapperClassName="tooltip-wrapper status-waiting-indicator">
-          <Icon color="red" id="yield" size="mini" />
-        </Tooltip>
-      );
+
+    if (item instanceof ServiceTree) {
```

Comment:
Couldn't we simplify this a bit and try to avoid those nested `if`-statements?
Didn't test this and not sure we actually need the `instance`-check if we could hit groups here, maybe it makes sense to have `getQueue` implemented on the `Group` struct to return `null` as we do other places so we don't need to check the instance type.
```js
class ServiceStatusWarning extends Component {
  getTooltip(content) {
    return (
      <Tooltip
        content={content}
        width={250}
        wrapText={true}
        wrapperClassName="tooltip-wrapper status-waiting-indicator">
        <Icon color="red" id="yield" size="mini" />
      </Tooltip>
    );
  }

  render() {
    const {item} = this.props;

    if (getAppsWithDeclinedOffersFromServiceTree(item).length > 0) {
      const appsNoun = StringUtil.pluralize('service', appCount);

      return this.getTooltip(`DC/OS is waiting for resources and is unable to complete the deployment of ${appCount} ${appsNoun} in this group.`);
    }

    if (item instanceof Service &&
      shouldDisplayDeclinedOffersWarning(item.getQueue())) {
      const queue = item.getQueue();
      const waitingSince = DateUtil.strToMs(queue.since);
      const timeWaiting = Date.now() - waitingSince;

      return this.getTooltip(`DC/OS has been waiting for resources and is unable to complete this deployment for ${DateUtil.getDuration(timeWaiting, null)}.`);
    }

    return <noscript />;
  }
};
```